### PR TITLE
Feat/redirect banner

### PR DESCRIPTION
**Pull request by envin3**
Add banner in order to redirect to currently supported url.
Prevent swap in case of 0 balance.

image
image

### CODE REVIEW

File: src/components/molecules/popup/RedirectBanner.jsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    console.error('Error while retreiving url', _err.message)
+    return null
+  }
+}
+
+const RedirectBanner = () => {
+  const domain = getDomain()
+  return (
+    <>
+      {domain && domain.includes(DISMISSED_DOMAIN) ? (
```
[the remark this answers is not in the document]
Comment (envin3) [reply src/components/molecules/popup/RedirectBanner.jsx:48]: This is the point of the code. If ptokens.io is anywhere in the url it should suggest to move to the new url.
I would dismiss this alert as non relevant for this usecase.

File: src/components/molecules/popup/RedirectBanner.jsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            </StyledAlert>
+          </OuterContainerSwap>
+        </Row>
+      ) : null}
+    </>
+  )
+}
+
+RedirectBanner.propTypes = {
+  content: PropTypes.node,
```
Comment (oliviera9) [inline src/components/molecules/popup/RedirectBanner.jsx:68]: Unused prop?

File: src/components/molecules/popup/__test__/RedirectBanner.test.jsx
Comment (oliviera9) [inline src/components/molecules/popup/__test__/RedirectBanner.test.jsx:1]: Great

File: src/components/molecules/popup/__test__/RedirectBanner.test.jsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import { render, screen } from '@testing-library/react'
+import { describe, expect, it, vi } from 'vitest'
+
+import RedirectBanner from '../RedirectBanner'
+
+const expectedText = 'You are using an old domain that will be dismissed in the future.'
+
+describe('RedirectBanner', async () => {
+  test.each([
+    ['http://ptokens.io'],
```
Comment (oliviera9) [inline src/components/molecules/popup/__test__/RedirectBanner.test.jsx:11]: Add a test for `pnetworkprotocol.eth`!

File: src/components/molecules/popup/__test__/RedirectBanner.test.jsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    ['https://dapp.pnetworkprotocol.eth.limo/'],
+    ['ipns://dapp.pnetworkprotocol.eth']['https://dapp.pnetworkprotocol.eth/#/swap?asset=btc&from=btc&to=eth'],
+  ])('Should show the url popup only on specific url', async (url) => {
+    Object.defineProperty(window, 'location', {
+      value: {
+        href: url,
+      },
+    })
+    const { container } = render(<RedirectBanner />)
+    expect(container.firstChild).toBeNull()
```
Comment (oliviera9) [inline src/components/molecules/popup/__test__/RedirectBanner.test.jsx:37]: :+1

File: src/components/pages/swap/__test__/Swap.test.jsx [hunk 5]
```diff
@@ -60,6 +63,48 @@ describe('Swap', async () => {
...
+    await UserEvent.type(fromInput, '1')
+    expect(fromInput).toHaveAttribute('value', '1')
+    expect(toInput).toHaveAttribute('value', '-2')
+    expect(swapButton).toHaveTextContent(
+      `${swapAssets.find((_el) => _el.id === PNT_ON_ETH_MAINNET).symbol} balance is 0`
+    )
+    expect(swapButton).toBeDisabled()
+  })
+
+  it('Should continue with balance null', async () => {
```
Comment (oliviera9) [inline src/components/pages/swap/__test__/Swap.test.jsx:84]: Why should we continue if null?
How is it different from having balance == 0?

**Review by oliviera9 — COMMENTED** [review]
Just a few comments

File: src/components/pages/swap/__test__/Swap.test.jsx (hunk 5 above)
Comment (envin3) [reply src/components/pages/swap/__test__/Swap.test.jsx:84]: Balance null happens if you didn't load the wallet:
in this case continue means continue to connect wallet or to get deposit address if you have bitcoin selected.

File: src/components/molecules/popup/RedirectBanner.jsx (hunk 2 above)
Comment (envin3) [reply src/components/molecules/popup/RedirectBanner.jsx:68]: Yes copy paste error

**Review by oliviera9 — APPROVED** [review]
LGTM!

**Pull request merged**